src/cli/cli.cpp

```diff
@@ -4667,6 +4698,7 @@ otError Interpreter::ProcessNetworkDiagnostic(Arg aArgs[])
     {
         SuccessOrExit(error = otThreadSendDiagnosticGet(mInstance, &address, tlvTypes, count,
                                                         &Interpreter::HandleDiagnosticGetResponse, this));
+        SetCommandTimeout(5000);
```

Comment:
Is the `SetCommandTimeout()` and `mTimer` only used for the `otThreadSendDiagnosticGet()`? Do we intend to use it for other command later?

I see that we don't call `OutputResult()` in `HandleDiagnosticGetResponse()` anymore. So does it mean that we always wait for 5seconds for this command (even if we receive a response, we still wait for 5 seconds, is this correct?).

If it will only be used for this command,  can we consider adding the timeout in the `otThreadSendDiagnosticGet()` (and `NetworkDiagnostic` module) rather than in CLI?

I am ok if we leave such a change for a later PR. Just wanted to suggest it and get peoples ideas/thoughts on this.

Reply:
> Is the `SetCommandTimeout()` and `mTimer` only used for the `otThreadSendDiagnosticGet()`? Do we intend to use it for other command later?

Yes. I intend to use this single Timer for all commands that need timeouts, saving RAM size.
Another possible use case for command timeout is `joiner start`. It's also useful for `wait` command in https://github.com/openthread/openthread/pull/6979.

> I see that we don't call `OutputResult()` in `HandleDiagnosticGetResponse()` anymore. So does it mean that we always wait for 5seconds for this command (even if we receive a response, we still wait for 5 seconds, is this correct?).

Yes, that's correct. 

Currently `networkdiagnostic get` command always wait for 5s. We can add a `timeout` argument to `networkdiagnostic get` command when when need more controls over timeouts.

Reply:
SGTM. Thanks.